### ISSUE TITLE
Normalise OL Layer's naming inside Hajk

### ISSUE THREAD

**Issue by jacobwod** [hajkmap/Hajk#883]
When we add a `Layer` to the `Map`, we currently set some additional properties. Two of them are redundant: `name`, and `type` - where name is used mostly (and is, in fact a wrong term, as it's more of an `id`). 

In addition, in some cases we add a `Layer` to the `Map`, without supplying neither `name`, `type` nor `caption`. 

Here's an example for `[l.get("name"), l.get("type"), l.get("caption")]`:
![Skärmavbild 2021-10-21 kl  11 39 18](https://user-images.githubusercontent.com/110222/138252821-c5d93732-83b7-41d4-8b2f-08527d458e49.png)

In the example above, 29-30 should have a `name` added, 31 should have the `type` value moved to `name` (and all references in code must be changed, naturally). Finally, all names should follow the same casing, be it camelCase or kebab-case.

**Comment by Hallbergs** [comment]
Let's hold this off until we're rewriting the LayerSwitcher-plugin

**Comment by jacobwod** [comment]
While implementing #1086 I came into the conclusion that this is very much a pre-requirement for a successful implementation of #1086. Even if we don't fix the naming/caption confusion right away, **we should at least add a `type` property to all layers, with one of the following values: `system`, `layer`, `groupLayer`** (perhaps more?).

Without this #1086 gets unnecessarily complicated.

**Comment by jacobwod** [comment]
I dropped the `groupLayer` idea as we already had a `group` value which did the same. 

In addition, I dropped the `system` property, in favour of the existing `layerType` property. **However, it got moved from the `ol/layer` object itself and into it's "hidden" `values_` property. This way, we can use the `getter/setter` as intended for layers properties in OpenLayers.**

Now the system layers look way better: correct type, coherent name and (as a bonus) caption that can be used in the UI:
![Skärmavbild 2022-05-24 kl  14 26 22](https://user-images.githubusercontent.com/110222/170034289-61ff4cd7-31c6-4284-9249-a41d5319c693.png)
